contracts/test/SetupPaymentsLib.t.sol

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "forge-std/Test.sol";
+import "../script/utils/SetupPaymentsLib.sol";
+import "../script/utils/CoreDeploymentLib.sol";
+import "../script/utils/HelloWorldDeploymentLib.sol";
+import "@eigenlayer/contracts/interfaces/IRewardsCoordinator.sol";
+import "@eigenlayer/contracts/interfaces/IStrategy.sol";
+import "../script/DeployEigenLayerCore.s.sol";
+import "../script/HelloWorldDeployer.s.sol";
+import {StrategyFactory} from "@eigenlayer/contracts/strategies/StrategyFactory.sol";
+
+contract TestConstants {
+    uint256 constant NUM_PAYMENTS = 8;
+    uint256 constant NUM_TOKEN_EARNINGS = 1;
+    uint256 constant TOKEN_EARNINGS = 100;
+
+    address RECIPIENT = address(1);
+    address EARNER = address(2);
+    uint256 INDEX_TO_PROVE = 0;
+}
+
+contract SetupPaymentsLibTest is Test, TestConstants {
+    using SetupPaymentsLib for *;
+
+    IRewardsCoordinator public rewardsCoordinator;
+    IStrategy public strategy;
+    ERC20Mock public token;
+
+
+    function setUp() public {
+        rewardsCoordinator = new MockRewardsCoordinator();
+        token = new ERC20Mock();
+        strategy = new MockStrategy(token);
+
+
+    }
+
+
+    function testSubmitPaymentRoot() public {
```

Comment:
You'll need to pull in the latest changes from master. The changes to the MockERC20 will probably break this since it will now require an approval tx